### PR TITLE
Remove ic_tcp and ic_udp related codes

### PR DESCRIPTION
**Pull request by pengzhout**
Codes of ic_tcp and ic_udp are not maintained any more, ic_udpifc is
the only interconnect type for now. We keep the gp_interconnect_type
guc for backward compatibility.

### CODE REVIEW

**Comment by ivannovick** [comment]
+1 to a lot less code. NICE!

**Comment by hsyuan** [comment]
+1

**Comment by pengzhout** [comment]
@yydzero  @xiong-gang  @atris @eurekaka 

please review this.

File: src/backend/cdb/cdbconn.c [hunk 1]
```diff
@@ -301,6 +303,14 @@ cdbconn_doConnect(SegmentDatabaseDescriptor *segdbDesc,
 		nkeywords++;
 	}
 
+	if (cli_port > 0)
+	{
+
```
Comment (danielgustafsson) [inline src/backend/cdb/cdbconn.c]: Excessive whitespace.

File: src/backend/cdb/cdbgang.c [hunk 2]
```diff
@@ -796,9 +801,52 @@ buildGangDefinition(GangType type, int gang_id, int size, int content, char *por
 		elog(ERROR, "Not all primary segment instances are active and connected");
 		/* not reached */
 	}
+
+	if (newGangDefinition->size > 1)
+	{
+		if (!getFreePortForGang(&newGangDefinition->dummy_sock, &newGangDefinition->cli_port))
+			elog(ERROR, "Can not get a free tcp port for N-gang");
```
Comment (danielgustafsson) [inline src/backend/cdb/cdbgang.c]: s/Can not/Cannot/ to stick to current wording in GPDB

File: src/backend/cdb/motion/cdbmotion.c [hunk 3]
```diff
@@ -154,10 +154,10 @@ initMotionLayerStructs(MotionLayerState **mlStates)
 	if (Gp_role == GP_ROLE_UTILITY)
 		return;
 
-	if (Gp_interconnect_type == INTERCONNECT_TYPE_UDPIFC || Gp_interconnect_type == INTERCONNECT_TYPE_UDP)
+	if (Gp_interconnect_type == INTERCONNECT_TYPE_UDPIFC)
 		Gp_max_tuple_chunk_size = Gp_max_packet_size - sizeof(struct icpkthdr) - TUPLE_CHUNK_HEADER_SIZE;
-	else
-		Gp_max_tuple_chunk_size = Gp_max_packet_size - PACKET_HEADER_SIZE - TUPLE_CHUNK_HEADER_SIZE;		
+	else
```
Comment (danielgustafsson) [inline src/backend/cdb/motion/cdbmotion.c]: Whitespace

**Comment by danielgustafsson** [comment]
With only one option allowed in the GUC can't we just fail server startup on incorrect values and thus be able to get rid of all the tests like this that remain?
```C
if (Gp_interconnect_type == INTERCONNECT_TYPE_UDPIFC) { .. }
```
Seems we can make the code even clearer and less errorprone with that tests just in a single place.

**Comment by pengzhout** [comment]
Hi @danielgustafsson,
It's a good advice since we have only one interconnect type left. my only concern is i want to leave interface here in case a new type may be added some day. what do you think?

**Comment by danielgustafsson** [comment]
If we add a new type at some point then we'll re-add the checks or perhaps even a completely new API (the hypothetical thing we might add in the future perhaps doesn't fit in this API anyways or might open up for a completely different one, we don't know). Until then, clean and easily readable code is more important so I think the checks should be removed with just one left at the point of entry of the config.

**Comment by pengzhout** [comment]
ok, make sense, let's make code cleaner.

**Comment by pengzhout** [comment]
Thanks for review, merged and closed

**Pull request closed without merging**